src/main/java/com/microfocus/application/automation/tools/run/SonarOctaneListener.java

```diff
@@ -119,25 +134,23 @@ public boolean isPushCoverage() {
 	 * sonar plugin data
 	 *
 	 * @param run               current run
-	 * @param allConfigurations jenkins global configuration
 	 * @throws InterruptedException
 	 */
-	private String[] getSonarDetails(@Nonnull Run<?, ?> run, ExtensionList<GlobalConfiguration> allConfigurations, TaskListener listener) throws InterruptedException {
-		String[] serverDetails = new String[2];
+	private void initializeSonarDetails(@Nonnull Run<?, ?> run, TaskListener listener) throws InterruptedException {
 		// if one of the properties is empty, need to query sonar plugin from jenkins to get the data
-		if (sonarProjectKey.isEmpty() || sonarToken.isEmpty() || sonarServerUrl.isEmpty()) {
+		ExtensionList<GlobalConfiguration> allConfigurations = GlobalConfiguration.all();
+		if (sonarToken.isEmpty() || sonarServerUrl.isEmpty()) {
 			try {
 				if (allConfigurations != null) {
 					SonarHelper adapter = new SonarHelper(run, listener);
-					serverDetails[0] = sonarServerUrl.isEmpty() ? adapter.getServerUrl() : sonarServerUrl;
-					serverDetails[1] = sonarToken.isEmpty() ? adapter.getServerToken() : sonarToken;
+					setSonarServerUrl(sonarServerUrl.isEmpty() ? adapter.getServerUrl() : sonarServerUrl);
+					setSonarToken(sonarToken.isEmpty() ? adapter.getServerToken() : sonarToken);
 
 				}
 			} catch (Exception e) {
 				throw new InterruptedException("exception occurred while init sonar tracker for job " + run.getDisplayName() + " error message: " + e.getMessage());
```

Comment:
If the intention of this is failing the build, I am not sure if it's the right approach.

Reply:
This code is not running as part of build but as thread after job is finished. Its not intended to fail the build.